src/_P118_Itho.ino

```diff
@@ -244,134 +246,149 @@ boolean Plugin_118(byte function, struct EventStruct *event, String &string)
 	case PLUGIN_WRITE: {
 		String tmpString = string;
 		String cmd = parseString(tmpString, 1);
-		String param1 = parseString(tmpString, 2);
 			if (cmd.equalsIgnoreCase(F("STATE")))
 			{
-
-				if (param1.equalsIgnoreCase(F("1111")))
-				{
-					noInterrupts();
-					PLUGIN_118_rf.sendCommand(IthoJoin);
-					interrupts();
-					PLUGIN_118_rf.initReceive();
-					addLog(LOG_LEVEL_INFO, F("Sent command for 'join' to Itho unit"));
-					printWebString += F("Sent command for 'join' to Itho unit");
-					success = true;
-				}
-				if (param1.equalsIgnoreCase(F("9999")))
-				{
-					noInterrupts();
-					PLUGIN_118_rf.sendCommand(IthoLeave);
-					interrupts();
-					PLUGIN_118_rf.initReceive();
-					addLog(LOG_LEVEL_INFO, F("Sent command for 'leave' to Itho unit"));
-					printWebString += F("Sent command for 'leave' to Itho unit");
-					success = true;
-				}
-			  if (param1.equalsIgnoreCase(F("0")))
+				switch(event->Par1) {
+					case 1111:
+					{
+						noInterrupts();
+						PLUGIN_118_rf.sendCommand(IthoJoin);
+						interrupts();
+						PLUGIN_118_rf.initReceive();
+						addLog(LOG_LEVEL_INFO, F("Sent command for 'join' to Itho unit"));
+						printWebString += F("Sent command for 'join' to Itho unit");
+						success = true;
+						break;
+					}
+				 	case 9999:
```

Comment:
Can you at least make defines for these "magical" numbers?
Makes it easier to read what you're doing here.

Reply:
They are explained at the start of the plugin, but I can add some local comments.